### PR TITLE
Executors

### PR DESCRIPTION
**Pull request by Bulldog2740**

### CODE REVIEW

File: executors/ExecutorAppAll.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package executors;
+
+import java.util.concurrent.*;
```
Comment (Anderwerty) [inline executors/ExecutorAppAll.java]: Don't use a wild card in import. I have already written about it in chat. If you could not find, ask your teammates.

File: executors/ExecutorAppAll.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+public class ExecutorAppAll {
+    public static void main(String[] args) throws Exception {
```
Comment (Anderwerty) [inline executors/ExecutorAppAll.java]: You put everything into one main method. Please separate this code into parts( use several main methods)

File: executors/FactorialCalculator.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+public class FactorialCalculator implements Callable<Integer> {
+
+    private Integer number;
+
+    public FactorialCalculator(Integer number) {
+        this.number = number;
+    }
+
+    @Override
+    public Integer call() throws Exception {
```
Comment (Anderwerty) [inline executors/FactorialCalculator.java]: It is better to change Integer to Long. It is necessary to take into account the possible overflow.

File: executors/FactorialCalculator.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        int result = 1;
+        if ((number == 0) || (number == 1)) {
+            result = 1;
+        } else {
+            for (int i = 2; i <= number; i++) {
+                result *= i;
+                TimeUnit.MILLISECONDS.sleep(20);
+            }
+        }
+        System.out.println("Result for number - " + number + " -> " + result);
```
Comment (Anderwerty) [inline executors/FactorialCalculator.java]: One responsibility per method. This method shouldn't print this information.

File: executors/ForkPoolApp [hunk 5]
```diff
@@ -0,0 +1,46 @@
+package executor;
+
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.RecursiveTask;
+
+public class ForkPoolApp {
+    private static long numOfOperations = 10_000_000_000L;
```
Comment (Anderwerty) [inline executors/ForkPoolApp]: Should it be a constant?

File: executors/ForkPoolApp [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    static class MyFork extends RecursiveTask<Long> {
+        long from;
+        long to;
+
+        @Override
+        protected Long compute() {
+            if (to - from <= numOfOperations / numOfThreads) {
+                long j = 0;
```
Comment (Anderwerty) [inline executors/ForkPoolApp]: `j` is not a good name for a local variable, only for an iterable variable. But it is not iterable variable.

File: executors/ForkPoolApp [hunk 7]
```diff
@@ -0,0 +1,46 @@
+package executor;
```
Comment (Anderwerty) [inline executors/ForkPoolApp]: It looks like you add this file in web. Please use commits.

File: executors/ExecutorAppAll.java (hunk 2 above)
Comment (Bulldog2740) [reply executors/ExecutorAppAll.java]: i try to make 3 class and put  all in them...hope i understend your idea right

File: executors/FactorialCalculator.java (hunk 4 above)
Comment (Bulldog2740) [reply executors/FactorialCalculator.java]: оки

File: executors/FactorialCalculator.java (hunk 3 above)
Comment (Bulldog2740) [reply executors/FactorialCalculator.java]: не может сделать так.

Comment (Bulldog2740) [reply executors/FactorialCalculator.java]: сделал

File: executors/ExecutorAppAll.java (hunk 1 above)
Comment (Bulldog2740) [reply executors/ExecutorAppAll.java]: идея сама как-то меняет..когда много чего импортит

File: executors/ForkPoolApp (hunk 5 above)
Comment (Bulldog2740) [reply executors/ForkPoolApp]: йес

File: executors/ForkPoolApp (hunk 7 above)
Comment (Bulldog2740) [reply executors/ForkPoolApp]: sometimes it is wrong somthing and dont wont pushhhh((((

File: executors/ExecutorAppAll.java (hunk 1 above)
Comment (Anderwerty) [reply executors/ExecutorAppAll.java]: image
I have written who to solve this issue with IntelliJ idea.
Please use English to answer.

File: executors/ForkPoolApp (hunk 5 above)
Comment (Anderwerty) [reply executors/ForkPoolApp]: Naming is not according to convetion.  
https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names

File: executors/ForkPoolApp (hunk 6 above)
Comment (Anderwerty) [reply executors/ForkPoolApp]: This comment is not resolved.

File: executors/FactorialCalculator.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+    public FactorialCalculator(Long number) {
+        this.number = number;
+    }
+
+    @Override
+    public Long call() throws Exception {
+        long result = 1;
+        if ((number == 0) || (number == 1)) {
+            result = 1;
```
Comment (Anderwerty) [inline executors/FactorialCalculator.java]: You could return the result here. It is already is 1. In this case, you don't need to write `else` just loop.

File: executors/ForkPoolApp [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.RecursiveTask;
+
+public class ForkPoolApp {
+    private static long NUM_OF_OPERATIONS = 10_000_000_000L;
+    private static int numOfThreads = Runtime.getRuntime().availableProcessors();
+
+    public static void main(String[] args) {
```
Comment (Anderwerty) [inline executors/ForkPoolApp]: Please move logic and main method into separate classes.

File: executors/ForkPoolApp [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+        long before = System.currentTimeMillis();
+        ForkJoinPool pool = new ForkJoinPool(numOfThreads);
+        pool.invoke(new MyFork(0, NUM_OF_OPERATIONS));
+        long after = System.currentTimeMillis();
+        System.out.println(after - before);
+    }
+
+    static class MyFork extends RecursiveTask<Long> {
```
Comment (Anderwerty) [inline executors/ForkPoolApp]: MyFork does not show what this class for? Please think about a better name.

File: executors/ForkPoolApp [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+        long from;
+        long to;
+
+        @Override
+        protected Long compute() {
+            if (to - from <= NUM_OF_OPERATIONS / numOfThreads) {
+                long jeff = 0;
```
Comment (Anderwerty) [inline executors/ForkPoolApp]: jeff? What is` jeff`?

Comment (Bulldog2740) [reply executors/ForkPoolApp]: the name? i cant take words what i want?

Comment (Anderwerty) [reply executors/ForkPoolApp]: The name must reflect the essence of the variable

File: executors/ForkTask.java [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+    public ForkTask(long from, long to) {
+        this.from = from;
+        this.to = to;
+    }
+
+    @Override
+    protected Long compute() {
+        if (to - from <= NUM_OF_OPERATIONS / numOfThreads) {
+            long sumnum = 0;
```
Comment (Anderwerty) [inline executors/ForkTask.java]: https://mate-academy.github.io/style-guides/java/java.html#s5.2.5-non-constant-field-names

File: executors/FactorialCalculator.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+    private Long number;
+
+    public FactorialCalculator(Long number) {
+        this.number = number;
+    }
+
+    @Override
+    public Long call() throws Exception {
+        long result = 1;
+        if ((number == 0) || (number == 1)) {
```
Comment (Anderwerty) [inline executors/FactorialCalculator.java]: In this case. You don't need empty if block.

File: executors/ForkTask.java [hunk 14]
```diff
@@ -0,0 +1,34 @@
+package executor;
+
+import java.util.concurrent.RecursiveTask;
+
+public class ForkTask extends RecursiveTask<Long> {
+    private long from;
+    private long to;
+    public static final long NUM_OF_OPERATIONS = 10_000_000_000L;
+    public static int numOfThreads = Runtime.getRuntime().availableProcessors();
```
Comment (Anderwerty) [inline executors/ForkTask.java]: Should it be a constant?

File: executors/ThreadPoolExTest.java [hunk 15]
```diff
@@ -0,0 +1,21 @@
+package executor;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+public class ThreadPoolExTest {
+    public static void main(String[] args) {
+        ThreadPoolExecutor es =
```
Comment (Anderwerty) [inline executors/ThreadPoolExTest.java]: Why not in one line? 
`ThreadPoolExecutor es =new ThreadPoolExecutor(3, 6, 1, TimeUnit.SECONDS, new LinkedBlockingQueue<>());`
Or if it is too long 
```
ThreadPoolExecutor es =
new ThreadPoolExecutor(3, 6, 1, TimeUnit.SECONDS, new LinkedBlockingQueue<>());
```

File: executors/ForkTask.java [hunk 16]
```diff
@@ -0,0 +1,34 @@
+package executor;
+
+        import java.util.concurrent.RecursiveTask;
+
+public class ForkTask extends RecursiveTask<Long> {
+    private long from;
+    private long to;
+    public static final long NUM_OF_OPERATIONS = 10_000_000_000L;
```
Comment (Anderwerty) [inline executors/ForkTask.java]: Better put constants before fields. And add one blank line between constants and fields definition.

File: executors/ForkTask.java [hunk 17]
```diff
@@ -0,0 +1,34 @@
+package executor;
+
```
Comment (Anderwerty) [inline executors/ForkTask.java]: Please format code.

**Pull request closed without merging**